Fixes and improvements for the tree prototype

phorward opened the pull request.

tsteinruecken reviewed (changes requested):
Maybe split this one into separate PRs?
- Changing the API on onEdit etc (idc)
- Changing the API of checkSkelType (why?)
- The issues with deleteRecursive should have been already fixed (and some of the altered conditions look bogus)

phorward commented:
Hello @tsteinruecken,

I've updated the pull request and answer your questions.

> Changing the API on onEdit etc (idc)

This PR entirely fixes the Tree module and cleans it up, as all functions are called and behaved equally.

> Changing the API of checkSkelType (why?)

`_checkSkelType` quarantees that the tree type is valid and returns it in lower-case order for further usage.

> The issues with deleteRecursive should have been already fixed (and some of the altered conditions look bogus)

I fixed the condition you mean in [cd9d7b7](https://github.com/viur-framework/viur-core/pull/381/commits/cd9d7b7be594eb92f7cea15feab8acfd8a145d13). Everything else are valid renamings, as the pull request is based on the previous develop branch.
